src/hotspot/share/cds/filemap.cpp

```diff
@@ -1721,9 +1721,9 @@ bool FileMapInfo::remap_shared_readonly_as_readwrite() {
   // This path should not be reached for Windows; see JDK-8222379.
   assert(WINDOWS_ONLY(false) NOT_WINDOWS(true), "Don't call on Windows");
   // Replace old mapping with new one that is writable.
-  char *base = os::map_memory(_fd, _full_path, r->file_offset(),
-                              addr, size, false /* !read_only */,
-                              r->allow_exec());
+  char *base = os::map_memory(_fd, _full_path, r->file_offset(),                      //file info
+                              addr, size,                                             //memory info
+                              false /* !read_only */, r->allow_exec(), mtClassShared); // flags
```

Comment:
Your new comments are not aligned properly.

However, I still don't think you should do this structural change. It makes the code inconsistent with the code below. If you do want to do it, I think you should change the rest of the code as well. But I don't think you should do that in this RFE, so I'd prefer if you just change this to:
```suggestion
  char *base = os::map_memory(_fd, _full_path, r->file_offset(),
                              addr, size, false /* !read_only */,
                              r->allow_exec(), mtClassShared);
```

Reply:
Done.